Task 3: Abilities, Inventory and Ui refactoring

gabrielcicero45 opened the pull request:
[Gravação de tela de 02-02-2023 13:46:51.webm](https://user-images.githubusercontent.com/42891342/216388204-8bf7abc2-427c-4cf9-bae0-eaade6acf525.webm)

gabrielcicero45 committed:
feat: Abilities,Invetory and UI changes

gabrielcicero45 committed:
refactor: Refactoring tables and Adding AbilitiesList

julialbq commented on src/components/Attributes/Attributes.js:
```diff
@@ -1,15 +1,35 @@
-      <div className="sheet__item">Charisma: {charisma}</div>
-    </>
-  )
+    <table className="sheet__item">
```
Try not to use the tag table for style, just when you are rendering data as a table

julialbq commented on src/components/Ability/Ability.js:
```diff
@@ -0,0 +1,11 @@
+
+export default function Ability({name,description,manaCost}) {
+  return (
+    <tr>
```
Try not to use the tag table for style, just when you are rendering data as a table

julialbq commented on src/components/Inventory/Inventory.js:
```diff
@@ -0,0 +1,43 @@
+    totalWeight += Object.values(item)[0].weight * item.amount;
+  });
+  return (
+    <table>
```
Try not to use the tag table for style, just when you are rendering data as a table

julialbq commented on src/components/InventoryItem/InventoryItem.js:
```diff
@@ -0,0 +1,15 @@
+  const name = Object.keys(item)[0].replace('_', ' ')
+  const {description,weight,type} = Object.values(item)[0];
+  return (
+    <tr>
```
Try not to use the tag table for style, just when you are rendering data as a table

gabrielcicero45 committed:
style: Improving Readability
- Separates the const `player` into player.js file
- Removes the xp/level rendering logic
- Adds animations to the axe assets

negreirosleo commented on src/components/Attributes/Attributes.js:
```diff
@@ -1,15 +1,35 @@
 
-export default function Attributes({attributes}) {
-  const {strength,dexterity,agility,intelligency,wisdom,charisma} = attributes
+export default function Attributes({ attributes }) {
+  const { strength, dexterity, agility, intelligency, wisdom, charisma } =
+    attributes;
```
```suggestion
export default function Attributes({ strength, dexterity, agility, intelligency, wisdom, charisma }) {
```

negreirosleo replied:
Outra forma de fazer isso de maneira mais programática
```js
Object.entries(attributes).map(
  ([name, value]) => <td className="sheet__item">{name}: {value}</td>
)
```

Dessa forma transformamos nosso objeto em um array com o nome e o valor do attributo e renderizamos com apenas um map!

negreirosleo commented on src/components/Inventory/Inventory.js:
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import InventoryItem from "../InventoryItem/InventoryItem";
+export default function Inventory({ inventory, strength }) {
+  let totalWeight = 0;
+  inventory.forEach((item) => {
+    totalWeight += Object.values(item)[0].weight * item.amount;
+  });
```
textbook reduce:
```js
  const totalWeight = inventory.reduce((accumulator, currentValue) => {
    return accumulator + Object.values(currentValue)[0].weight * currentValue.amount;
  }, 0);
  ```

negreirosleo commented on src/components/Inventory/Inventory.js:
```diff
@@ -0,0 +1,43 @@
+          <td colSpan={2}></td>
+          <td>Weight Status:</td>
+          <td colSpan={2}>
+            {totalWeight > strength * 2 ? "Overweight" : "Ok"}
```
Seria legal ter esse strength * 2 numa variavel

gabrielcicero45 committed:
refactor: Inventory refactor and Removing unnecessary rules

gabrielcicero45 commented on src/components/Sheet/Sheet.js:
```diff
@@ -1,52 +1,48 @@
+      <div className="sheet__container">
+        <div className="sheet__column">
+          <RpgClass rpgClass={rpgClass} />
+          <Name name={name} surname={surmane} />
```
```suggestion
          <Name name={name} surname={surname} />
```

gabrielcicero45 commented on src/components/Sheet/Sheet.js:
```diff
@@ -1,52 +1,48 @@
+import Inventory from "../Inventory/Inventory";
+import AbilitiesList from "../AbilitiesList/AbilitiesList";
+export default function Sheet({ player }) {
+  const { name, surmane, health, mana, level, xp, attributes, rpgClass, abilities, inventory } =
```
```suggestion
  const { name, surname, health, mana, level, xp, attributes, rpgClass, abilities, inventory } =
```

gabrielcicero45 committed:
Update src/components/Sheet/Sheet.js

gabrielcicero45 committed:
Update src/components/Sheet/Sheet.js
